test/e2e/backup_restore_test.go

```diff
@@ -0,0 +1,314 @@
+		15*time.Second,
+		func(statusCode int, body string) bool {
+			return statusCode == 200
+		},
+	)
+
+	//set test data path and documents to be loaded for tests
+	docPath := "../test_data/bkup_restore_reqs/"
+	docName := "testOne.xml"
+	docNameTwo := "testTwo.xml"
+
+	tunnel8000 := k8s.NewTunnel(kubectlOptions, k8s.ResourceTypePod, podName, 8000, 8000)
+	defer tunnel8000.Close()
+	tunnel8000.ForwardPort(t)
+	qConsoleEndpoint := fmt.Sprintf("http://%s/v1/documents?database=Documents&uri=%s", tunnel8000.Endpoint(), docName)
+	fmt.Println(qConsoleEndpoint)
+
+	client := req.C().
+		SetCommonDigestAuth(username, password).
+		SetCommonRetryCount(10).
+		SetCommonRetryFixedInterval(10 * time.Second)
+
+	result, err := PutDocs(docPath, docName, client, qConsoleEndpoint)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	assert.Equal(t, "Created", result)
+
+	qConsoleEndpoint1 := fmt.Sprintf("http://%s/v1/documents?database=Documents&uri=%s", tunnel8000.Endpoint(), docNameTwo)
+	fmt.Println(qConsoleEndpoint1)
+
+	result, err = PutDocs(docPath, docNameTwo, client, qConsoleEndpoint1)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	assert.Equal(t, "Created", result)
+
+	qConsoleEndpoint2 := fmt.Sprintf("http://%s/v1/documents?database=Documents&uri=%s&uri=%s", tunnel8000.Endpoint(), docName, docNameTwo)
+	fmt.Println(qConsoleEndpoint2)
+
+	// verify both docs are loaded in Documents DB
+	result, err = GetDocs(client, qConsoleEndpoint2, "multipart/mixed")
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	if !strings.Contains(string(result), "<b>two</b>") && !strings.Contains(string(result), "<a>one</a>") {
+		t.Errorf("Both docs are loaded")
+	}
+
+	tunnel8002 := k8s.NewTunnel(kubectlOptions, k8s.ResourceTypePod, podName, 8002, 8002)
+	defer tunnel8002.Close()
+	tunnel8002.ForwardPort(t)
+	manageEndpoint := fmt.Sprintf("http://%s/manage/v2/databases/Documents", tunnel8002.Endpoint())
+	fmt.Println(manageEndpoint)
+
+	bkupReq := &BackupRestoreReq{
+		Operation:       "backup-database",
+		BackupDir:       "/tmp",
+		IncludeReplicas: "true"}
+	bkupReqRes, _ := json.Marshal(bkupReq)
+
+	//backup Documents DB
+	result, err = RunRequests(client, string(bkupReqRes), manageEndpoint)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	jobID := (gjson.Get(result, `job-id`))
+	hostName := (gjson.Get(result, `host-name`))
+
+	bkupStatusReq := &BackupRestoreStatusReq{
+		Operation: "backup-status",
+		JobID:     jobID.String(),
+		HostName:  hostName.String()}
+	bkupStatusReqRes, _ := json.Marshal(bkupStatusReq)
+
+	//get status of backup job
+	result, err = RunRequests(client, string(bkupStatusReqRes), manageEndpoint)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	bkupStatus := (gjson.Get(result, `status`)).Str
+
+	//verify backup is completed
+	assert.Equal(t, "completed", bkupStatus)
+
+	//delete a document from Documents DB
+	result, err = DeleteDocs(client, qConsoleEndpoint1)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	assert.Equal(t, "Deleted", result)
```

Comment:
Can we have some assert statement to verify if the documents are not present using endpoint after deletion along with validating the status ?

Reply:
Status "Deleted" is returned by function only when API: DELETE /v1/documents returns a success response code 204 when the document is deleted from DB.

Reply:
Just as double check wanted to query and also confirm if document is deleted before restore rather than just status. But if we are confident that document will be deleted it should be okay